### ISSUE TITLE
ESP32 with ESP-MDF: In Field/deployment Corruptions Issue - Need Help (IDFGH-11146)

### ISSUE THREAD

**Issue by karthiksb0307** [espressif/esp-mdf#341]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [ ] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### General issue report

I'm encountering a perplexing issue with my ESP32 devices running ESP-IDF 4.3 with ESP-MDF. During factory testing, all devices perform flawlessly. However, after being deployed in the field for a few weeks, some of them become corrupted or malfunction.

The problem manifests as device corruption or malfunctioning after several weeks of field operation. This issue occurs sporadically and does not affect all devices.

[14:54:20:297] <0x1b>[0;32mI (13) boot: ESP-IDF v4.3-beta3 2nd stage bootloader<0x1b>[0m␍␊
[14:54:20:306] <0x1b>[0;32mI (13) boot: compile time 17:22:21<0x1b>[0m␍␊
[14:54:20:311] <0x1b>[0;32mI (13) boot: chip revision: 1<0x1b>[0m␍␊
[14:54:20:311] <0x1b>[0;32mI (15) boot_comm: chip revision: 1, min. bootloader chip revision: 0<0x1b>[0m␍␊
[14:54:20:320] <0x1b>[0;32mI (22) qio_mode: Enabling default flash chip QIO<0x1b>[0m␍␊
[14:54:20:326] <0x1b>[0;32mI (27) boot.esp32: SPI Speed : 40MHz<0x1b>[0m␍␊
[14:54:20:332] <0x1b>[0;32mI (32) boot.esp32: SPI Mode : QIO<0x1b>[0m␍␊
[14:54:20:337] <0x1b>[0;32mI (37) boot.esp32: SPI Flash Size : 4MB<0x1b>[0m␍␊
[14:54:20:337] <0x1b>[0;32mI (41) boot: Enabling RNG early entropy source...<0x1b>[0m␍␊
[14:54:20:344] <0x1b>[0;32mI (47) boot: Partition Table:<0x1b>[0m␍␊
[14:54:20:351] <0x1b>[0;32mI (50) boot: ## Label Usage Type ST Offset Length<0x1b>[0m␍␊
[14:54:20:357] <0x1b>[0;32mI (57) boot: 0 nvs WiFi data 01 02 00009000 00004000<0x1b>[0m␍␊
[14:54:20:363] <0x1b>[0;32mI (65) boot: 1 otadata OTA data 01 00 0000d000 00002000<0x1b>[0m␍␊
[14:54:20:374] <0x1b>[0;32mI (72) boot: 2 phy_init RF data 01 01 0000f000 00001000<0x1b>[0m␍␊
[14:54:20:379] <0x1b>[0;32mI (80) boot: 3 ota_0 OTA app 00 10 00010000 001e0000<0x1b>[0m␍␊
[14:54:20:385] <0x1b>[0;32mI (87) boot: 4 ota_1 OTA app 00 11 001f0000 001e0000<0x1b>[0m␍␊
[14:54:20:392] <0x1b>[0;32mI (95) boot: 5 coredump Unknown data 01 03 003d0000 00010000<0x1b>[0m␍␊
[14:54:20:399] <0x1b>[0;32mI (102) boot: 6 reserved Unknown data 01 fe 003e0000 00020000<0x1b>[0m␍␊
[14:54:20:406] <0x1b>[0;32mI (110) boot: End of partition table<0x1b>[0m␍␊
[14:54:20:411] <0x1b>[0;32mI (114) boot_comm: chip revision: 1, min. application chip revision: 0<0x1b>[0m␍␊
[14:54:20:417] <0x1b>[0;32mI (121) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=47c78h (294008) map<0x1b>[0m␍␊
[14:54:20:510] <0x1b>[0;32mI (225) esp_image: segment 1: paddr=00057ca0 vaddr=3ffbdb60 size=05c5ch ( 23644) load<0x1b>[0m␍␊
[14:54:20:519] <0x1b>[0;32mI (234) esp_image: segment 2: paddr=0005d904 vaddr=40080000 size=02714h ( 10004) load<0x1b>[0m␍␊
[14:54:20:529] <0x1b>[0;32mI (238) esp_image: segment 3: paddr=00060020 vaddr=400d0020 size=12b638h (1226296) map<0x1b>[0m␍␊
[14:54:20:922] <0x1b>[0;32mI (637) esp_image: segment 4: paddr=0018b660 vaddr=40082714 size=1bd00h (113920) load<0x1b>[0m␍␊
[14:54:20:964] <0x1b>[0;32mI (680) esp_image: segment 5: paddr=001a7368 vaddr=50000000 size=00010h ( 16) load<0x1b>[0m␍␊
[14:54:20:973] <0x1b>[0;31mE (680) esp_image: Checksum failed. Calculated 0x67 read 0x47<0x1b>[0m␍␊
[14:54:20:983] <0x1b>[0;31mE (683) boot: OTA app partition slot 0 is not bootable<0x1b>[0m␍␊
[14:54:20:989] <0x1b>[0;31mE (689) esp_image: image at 0x1f0000 has invalid magic byte<0x1b>[0m␍␊
[14:54:20:997] <0x1b>[0;33mW (695) esp_image: image at 0x1f0000 has invalid SPI mode 255<0x1b>[0m␍␊
[14:54:21:003] <0x1b>[0;33mW (702) esp_image: image at 0x1f0000 has invalid SPI size 15<0x1b>[0m␍␊
[14:54:21:010] <0x1b>[0;31mE (708) boot: OTA app partition slot 1 is not bootable<0x1b>[0m␍␊
[14:54:21:017] <0x1b>[0;31mE (714) boot: No bootable app partitions in the partition table<0x1b>[0m␍␊

**Comment by wujiangang** [comment]
Yes, you can upload both of them here, so we can help analyze them.

**Comment by karthiksb0307** [comment]
![image](https://github.com/espressif/esp-idf/assets/58732434/bf97778b-4ad2-447d-8451-2e272e79a098)
the only difference between the original and corrupted is at at start address of app binary, "0xe9" changed to "0x00" as seen in the picture.

**Comment by karthiksb0307** [comment]
> ![image](https://user-images.githubusercontent.com/58732434/279359888-bf97778b-4ad2-447d-8451-2e272e79a098.png) the only difference between the original and corrupted is at at start address of app binary, "0xe9" changed to "0x00" as seen in the picture.

seems like magic byte is getting corrupted.

**Comment by tonyshuang** [comment]
您好，信件已收到。
